sdk/azcore/etag_test.go

```diff
@@ -0,0 +1,224 @@
+// +build go1.13
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azcore
+
+import (
+	"testing"
+)
+
+func TestEtagEquals(t *testing.T) {
+	e1 := NewETag("tag")
+	if e1.String() != "tag" {
+		t.Fatalf("ETag values are not equal")
+	}
+
+	e2 := NewETag("\"tag\"")
+	if e2.String() != "\"tag\"" {
+		t.Fatalf("ETag values are not equal")
+	}
+
+	e3 := NewETag("W/\"weakETag\"")
+	if e3.String() != "W/\"weakETag\"" {
```

Comment:
No `testify/require` allowed in azcore?

Reply:
It's not currently there but I'm going to add it because this is too much repeated code for tests